crates/libcontainer/src/container/container_start.rs

```diff
@@ -59,11 +59,6 @@ impl Container {
             })?;
         }
 
-        unistd::chdir(self.root.as_os_str()).map_err(|err| {
-            tracing::error!("failed to change directory to container root: {}", err);
-            LibcontainerError::OtherSyscall(err)
-        })?;
-
```

Comment:
Would there be any issues with the hook behavior by running this? I checked the spec once, and I don't think it requires to run the hooks in same dir as the container, only in the same namespace, but just to confirm.

Reply:
Good catch! You are right, thanks.
444146eb

Reply:
Hey, I think we need to re-change the cwd back to the original one after the `run_hooks` call, so that the dir is correct? Otherwise now the dir will change when hooks are there, and won't when they're not, it'll be inconsistent behavior.

Also, When I originally commented , I was talking about the hooks invocation code below this, [line 77->85.](https://github.com/containers/youki/pull/2780/files#diff-e21c19897fbff6f13bb303b180c4879d40d96f5a30a65497eb4f605fc3fb8341L78-L83)

Given that the existing invocation of hooks above  [line 52->60](https://github.com/containers/youki/pull/2780/files#diff-e21c19897fbff6f13bb303b180c4879d40d96f5a30a65497eb4f605fc3fb8341R56-R63) didn't have chdir before, I don't think it is needed now?

Also my concern while raising this point was that I'm not sure if hooks even need the chdir, as the spec says `in the same namespace` which does include the mount namespace, but does not say anything about the cwd if I understand it correctly.

Reply:
524ae7a8

You are probably right, but if we delete it, we must be responsible for the migration guide or something. So I don't prefer introducing such breaking change in this PR.
If we do, we should create another PR.

> Also my concern while raising this point was that I'm not sure if hooks even need the chdir, as the spec says in the same namespace which does include the mount namespace, but does not say anything about the cwd if I understand it correctly.

Reply:
Hey, I agree that we should not have breaking changes here.
But I think I'm still not able to convey my concern properly - The changes you have done are around the pre-start hooks, which are marked deprecated. I don't think this hook invocation was running in the root dir before, so for these hooks, these changes are breaking. On the other hand, there is the invocation of post-start hook _after_ the currently removed chdir which should be added between two chdir calls. Let me know if you still have questions, maybe I can just make a patch PR to you fork to make it better understandable.

Reply:
I opened https://github.com/utam0k/youki/pull/528 on your fork.

Reply:
Ops, you are right. I misunderstood the point you pointed out. Is it enough?
8de223377e21d5b1e9f1ce3c5d56e6d22f60e601